crates/services/gas_price_service/src/fuel_gas_price_updater.rs

```diff
@@ -104,6 +112,19 @@ pub trait L2BlockSource: Send + Sync {
     async fn get_l2_block(&mut self, height: BlockHeight) -> Result<BlockInfo>;
 }
 
+#[derive(Debug, Default)]
+pub struct DaCommitDetails {
+    pub l2_block_range: core::ops::Range<u32>,
+    pub blob_bytes: u32,
+    pub blob_cost_wei: u32,
+    pub partial_block_heights: Option<[u32; 2]>,
+}
+
+#[async_trait::async_trait]
+pub trait DaCommitSource: Send + Sync {
+    async fn get_da_commit_details(&mut self) -> Result<DaCommitDetails>;
```

Comment:
I think this should actually be sync, not `async`. This is because we want it to return immediately every time. Usually it will return an empty range. So maybe this should also be `Option` to capture that?

Reply:
addressed in 9d68aa1f2